### PR TITLE
Add test case for function regression

### PR DESCRIPTION
**Pull request by zhulinJulia24**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -84,7 +85,9 @@ jobs:
           make -j$(nproc) && make install
       - name: Install lmdeploy
         run: |
-          python3 -m pip install packaging protobuf transformers_stream_generator
+          python3 -m pip install packaging protobuf transformers_stream_generator transformers==4.33.0
+          wget  https://github.com/Dao-AILab/flash-attention/releases/download/v2.3.6/flash_attn-2.3.6+cu118torch2.0cxx11abiFALSE-cp38-cp38-linux_x86_64.whl
```
Comment (RunningLeon) [inline .github/workflows/test.yml:89]: # manually install flash attn
          python3 -m pip install /root/packages/flash_attn-2.3.6+cu118torch2.0cxx11abiFALSE-cp38-cp38-linux_x86_64.whl

File: autotest/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+Run pytest command line with case filtering through -m flag. eg: `-m internlm_chat_7b` Filter cases related to internlm_chat_7b. The corresponding results will be stored in the `allure-results` directory.
+
+```bash
+pytest autotest -m internlm_chat_7b --clean-alluredir --alluredir=allure-results
+```
+
+After the test cases are executed, you can generate the test report by using the following command. The generated report will be stored in `allure-reports`.
+
+```bash
```
Comment (RunningLeon) [inline autotest/README.md:21]: add installation of allure: https://allurereport.org/docs/gettingstarted-installation/#install-via-the-system-package-manager-for-linux
```
wget https://github.com/allure-framework/allure2/releases/download/2.25.0/allure_2.24.1-1_all.deb
sudo apt-get install -y openjdk-8-jre-headless
sudo dpkg -i ./allure_2.24.1-1_all.deb
```

File: autotest/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+| w4a16 quantization | test_order1_w4_quantization |
+| convert            | test_order2_convert         |
+| cli command chat   | test_order3_command_chat    |
+| hf command chat    | test_order3_hf_command_chat.py    |
+| restful_api chat   | test_order4_restful_chat    |
+
+The modules and models currently covered by the test cases are listed below:
+
+| Models                                                                     | w4a16 quantization | convert | cli command chat | restful_api chat | hf command chat  |
+| -------------------------------------------------------------------------- | ------------------ | ------- | ---------------- | ---------------- | ---------------- |
```
Comment (RunningLeon) [inline autotest/README.md]: could make the columns center-aligned by changing to 
```| -------------------------------------------------------------------------- | :------------------: | :-------: | :----------------: | :----------------: | :----------------: |```

**Comment by zhulinJulia24** [comment]
fix pr comments and lint errors.

File: autotest/utils/run_client_chat.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
+import os
+from subprocess import PIPE, Popen
+
+from utils.rule_condition_assert import assert_result
+
+
+def commandLineTest(config, case, case_info, model, type, extra):
```
Comment (RunningLeon) [inline autotest/utils/run_client_chat.py:7]: could make the naming style of vars, funs, classes consistent with this repo.

File: autotest/README.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+| :----------------: | :----------------------------: |
+| w4a16 quantization |  test_order1_w4_quantization   |
+|      convert       |      test_order2_convert       |
+|  cli command chat  |    test_order3_command_chat    |
+|  hf command chat   | test_order3_hf_command_chat.py |
+|  restful_api chat  |    test_order4_restful_chat    |
+
+The modules and models currently covered by the test cases are listed below:
+
+|                                   Models                                   | w4a16 quantization | convert | cli command chat | restful_api chat | hf command chat |
```
Comment (RunningLeon) [inline autotest/README.md:64]: all models do not have any test for restful_api chat ? why remove it?

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -150,10 +151,12 @@ jobs:
           echo "CONTAINER_ID=$CONTAINER_ID"  >> $GITHUB_ENV
       - name: Build lmdeploy from source
         run: |
-          docker exec $CONTAINER_ID cp -r ../../build build
+          docker exec $CONTAINER_ID mkdir build
```
Comment (RunningLeon) [inline .github/workflows/test.yml:154]: test_triton failed. It will be fixed in another PR.

File: autotest/README.md (hunk 5 above)
Comment (zhulinJulia24) [reply autotest/README.md:64]: > restful_api

after fix proxy, I will add it with restful case added

File: .github/workflows/test.yml (hunk 6 above)
Comment (zhulinJulia24) [reply .github/workflows/test.yml]: rollback down

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
